neoexchange/core/plots.py

```diff
@@ -316,52 +332,79 @@ def spec_plot(data_spec, analog_data, reflec=False):
+        ref_plot.add_tools(hover)
+        ref_plot.legend.click_policy = 'hide'
+        ref_plot.legend.orientation = 'horizontal'
 
         # set axes
-        plot2.axis.axis_line_width = 2
-        plot2.axis.axis_label_text_font_size = "12pt"
-        plot2.axis.major_tick_line_width = 2
-        plot2.xaxis.axis_label = "Wavelength (Å)"
-        plot2.yaxis.axis_label = 'Reflectance Spectra (Normalized at 5500 Å)'
-
-        spec_plots["reflec_spec"] = plot2
+        ref_plot.axis.axis_line_width = 2
+        ref_plot.axis.axis_label_text_font_size = "12pt"
+        ref_plot.axis.major_tick_line_width = 2
+        ref_plot.xaxis.axis_label = "Wavelength (Å)"
+        ref_plot.yaxis.axis_label = 'Reflectance Spectra (Normalized at 5500 Å)'
+
+    # Build tools
+    if analog_data:
+        analog_labels = [a['label'] for a in analog_data]
+    else:
+        analog_labels = ["--None--"]
+
+    analog_select = Select(title="Analog", value=analog_labels[0], options=analog_labels)
+    frame_labels = list(map(str, range(1, len(data_spec)+1)))
+    frame_select = MultiSelect(title="Target Frames", value=frame_labels, options=frame_labels, height=49, width=100)
+
+    # JS Call back to change analog
+    js_analog_picker = get_js_as_text(js_file, "analog_select")
+    analog_select_callback = CustomJS(args=dict(analog_select=analog_select, frame_select=frame_select,
+                                                reflectance_sources=reflect_source_lists,
+                                                chosen_sources=reflect_source_prefs, plot=raw_plot, plot2=ref_plot,
+                                                lines=reflectance_lines, raw_analog=analog_source_final,
+                                                raw_analog_list=analog_sources_raw, raw_lines=raw_lines),
+                                      code=js_analog_picker)
+    analog_select.js_on_change('value', analog_select_callback)
+    frame_select.js_on_change('value', analog_select_callback)
+
+    selector_row = row(frame_select, analog_select)
+    layouts = {'raw_and_reflect': column(selector_row, row(ref_plot), row(raw_plot)),
+               'raw_only': column(selector_row, row(raw_plot)),
+               'raw_standard': column(row(raw_plot)),
+               'reflect_only': column(row(ref_plot))
+               }
 
     # Create script/div
-    if spec_plots:
-        script, div = components(spec_plots, CDN)
-    else:
-        return '', {"raw_spec": ''}
-    try:
-        for key in div.keys():
-            b = div[key].index('>')
-            div[key] = '{} name={}{}'.format(div[key][:b], key, div[key][b:])
-    except ValueError:
-        pass
-    return script, div
+    script, div = components(layouts[spec_type], CDN)
+
+    chunks = div.split("data-root-id=")
+    named_div = chunks[0] + f'name="spec_plot" data-root-id=' + chunks[1]
+
+    return script, named_div
 
 
 def datetime_to_radians(ref_time, input_time):
```

Comment:
Wonder if this should be in `time_subs.py` ?

Reply:
This is a piece of code for basically making hands on an analog clock. I don't think it would hurt it to be in `time_subs` but it's pretty specific to the following application.

Reply:
Fair enough, best to leave it where it is if it's not in general use (the githab review cutoff at the `datetime_to_radians` method name so I didn't get the full context)